pkg/component/controller/metrics.go

```diff
@@ -0,0 +1,359 @@
+	}
+
+	resp, err := j.scrapeClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("error collecting metrics from %s: %w", j.scrapeURL, err)
+	}
+	defer resp.Body.Close()
+
+	res := j.restClient.Post().AbsPath(j.pushURL()).Body(resp.Body).Do(ctx)
+	if res.Error() != nil {
+		return fmt.Errorf("error sending POST request for job %s: %w", j.name, res.Error())
+	}
+	return nil
+}
+
+func getClient(certFile, keyFile string) (*http.Client, error) {
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	transport.ResponseHeaderTimeout = time.Minute
+	tlsConfig := &tls.Config{InsecureSkipVerify: true}
+	transport.TLSClientConfig = tlsConfig
+
+	if certFile != "" && keyFile != "" {
+		cert, err := tls.LoadX509KeyPair(certFile, keyFile)
+		if err != nil {
+			return nil, err
+		}
+		tlsConfig.Certificates = []tls.Certificate{cert}
+	}
+	return &http.Client{
+		Transport: transport,
+		Timeout:   time.Minute,
+	}, nil
+}
+
+const pushGatewayTemplate = `
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: {{ .Namespace }}
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+    component: "pushgateway"
+    app: k0s-observability
+  name: k0s-pushgateway
+  namespace: {{ .Namespace }}
+  annotations:
+    {}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  labels:
+    component: "pushgateway"
+    app: k0s-observability
+  name: {{ .Name }}
+rules:
+  []
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  labels:
+    component: "pushgateway"
+    app: k0s-observability
+  name: {{ .Name }}
+subjects:
+  - kind: ServiceAccount
+    name: {{ .Name }}
+    namespace: {{ .Namespace }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: {{ .Name }}
+---
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    prometheus.io/probe: pushgateway
+  labels:
+    component: "pushgateway"
+    app: k0s-observability
+  name: {{ .Name }}
+  namespace: {{ .Namespace }}
+spec:
+  ports:
+    - name: http
+      port: 9091
+      protocol: TCP
+      targetPort: 9091
+  selector:
+    component: "pushgateway"
+    app: k0s-observability
+  type: "ClusterIP"
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    component: "pushgateway"
+    app: k0s-observability
+  name: {{ .Name }}
+  namespace: {{ .Namespace }}
+spec:
+  selector:
+    matchLabels:
+      component: "pushgateway"
+      app: k0s-observability
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        component: "pushgateway"
+        app: k0s-observability
+    spec:
+      serviceAccountName: {{ .Name }}
+      tolerations:
+        - key: "CriticalAddonsOnly"
+          operator: "Exists"
+        - key: "node-role.kubernetes.io/master"
+          operator: "Exists"
+          effect: "NoSchedule"
+      containers:
+        - name: prometheus-pushgateway
+          image: "dmathai/prom-pushgateway-ttl:1.4.0-ttl"
```

Comment:
Should we handle this image as we do others (have it configurable)? And should we include this into airgap bundle too?

One downside on this is that is only available for amd64:
```
$ manifest-tool inspect dmathai/prom-pushgateway-ttl:1.4.0-ttl
dmathai/prom-pushgateway-ttl:1.4.0-ttl: manifest type: application/vnd.docker.distribution.manifest.v2+json
      Digest: sha256:a815e2c8aac8336beaf32f854009b33fe54d4fe2dc7fb30b2701a72c35b9a431
Architecture: amd64
          OS: linux
    # Layers: 4
      layer 1: digest = sha256:e5d9363303ddee1686b203170d78283404e46a742d4c62ac251aae5acbda8df8
      layer 2: digest = sha256:3430c2c42129e987db5d5c5434ea73f0cbb44b931719cb2072203fcc05617d54
      layer 3: digest = sha256:a09edd3728450ef9d785d66237dfa943554b8f518d025fd463094d165c3c8774
      layer 4: digest = sha256:547263457d796dc5c03b1846252566f39426aff85d3c5caed3405f5800bc64a4
```

Reply:
Yeah, fair enough. We also need to push the image to our own registry, I think.

Reply:
Make it configurable and it uses our image by default now